docs/proposals/hpa/federated-hpa-v2.md

```diff
@@ -0,0 +1,809 @@
+```
+target cpu util 30%
+min replica 3
+max replica 100
+```
+Suddenly, one of the member clusters which my application running on can't scale up new pods. Unfortunately, a request burst is coming into the application. The CPU util of pods becomes higher than 30%. It will need 100 Pods totally to take the request burst. I hope the Karmada FederatedHPA can scale up new pods in other healthy clusters with available resource.
+
+#### Scale workloads across multi-clusters with rich policies
+As a administrator, I have some clusters in my local data center as well as some clusters in the cloud. I want to scale  up the works in local clusters, if the resource is not enough, then scale up the workloads in the cloud clusters. Besides this, it may include other scenarios, I want Karmada FederatedHPA have rich policies to enable them.
+
+#### Tolerate the failure of Karmada control plane
+As an administrator, if the Karmada control plane is down, but there are many applications running on the platform heavily depend on the HPA to handle the unpredictable burst of requests. So I want scaling should still work to a certain extent.
+
+#### Scale workloads across multi-clusters with time delay
+As a platform developer, I may install cluster-autoscaler in some clusters, which means it will add nodes into the clusters when the resources is insufficient. If so, scale across multi-clusters is not necessary if cluster-autoscaler works. I want Karmada FederatedHPA could execute scaling across multi-clusters only when cluster-autoscaler doesn't work for specific time.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go into as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+1. The workloads/pods in different member clusters selected by the same HPA CR/resource share the load of the application equally. For example, 10 pods of the application are spread into two member clusters with distribution `cluster1: 3 pods, cluster2: 7 pods`, so the 3 pods in cluster1 take 3/10 of total requests and 7 pods in cluster2 take 7/10 of total requests. Scenarios that don't meet the restriction are not considered in this proposal.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate them? 
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+1. This proposal doesn't consider cluster disaster migration for HPA, which should be done in another proposal.
+1. When assigning the FederatedHPA, we use availableReplicas as the key factor, it may cause inaccurate HPA assignments.
+
+## Design Details
+### Architecture
+![Karmada Federated HPA Architecture](./statics/karmadafederatedhpa-arch-v2.drawio.png)
```

Comment:
According to this image, I guess scheduler does not take any effect on Federated HPA workload, and all scaling result is based on the decision made by HPA controller in member clusters, right?

Reply:
Correct, the core concept in this proposal is: when a member cluster could scale up the workloads, FederatedHPA controller should not care about it, but when it can't(reached maxReplicas or pending, or other reason), FederatedHPA controller should select one and scale it up.